test/test_helpers.py

```diff
@@ -0,0 +1,118 @@
+import numpy as np
+from covidactnow.datapublic.common_fields import CommonFields
+from covidactnow.datapublic.common_fields import FieldName
+from covidactnow.datapublic.common_fields import PdFields
+
+from libs.datasets import timeseries
+from libs.pipeline import Region
+
+
+# This is a totally bogus fips/region/location that we've been using as a default in some test
+# cases. It is factored out here in an attempt to reduce how much it is hard-coded into our source.
+DEFAULT_FIPS = "97222"
+DEFAULT_REGION = Region.from_fips(DEFAULT_FIPS)
+
+
+class TimeseriesLiteral(UserList):
+    """Represents a timeseries literal, a sequence of floats and provenance string."""
+
+    def __init__(
+        self, ts_list, *, provenance: str = "",
+    ):
+        super().__init__(ts_list)
+        self.provenance = provenance
+
+
+def build_dataset(
+    metrics: Mapping[Region, Mapping[FieldName, Union[Sequence[float], TimeseriesLiteral]]],
+    *,
+    start_date="2020-04-01",
+    dropna=True,
+) -> timeseries.MultiRegionDataset:
+    """Returns a dataset for multiple regions and metrics. Each sequence of values represents a
+    timeseries metric with identical length. provenance information can be set for a metric by
+    using a TimeseriesLiteral. Timeseries without any real values are dropped.
+    """
+    # From https://stackoverflow.com/a/47416248. Make a dictionary listing all the timeseries
+    # sequences in metrics.
+    loc_var_seq = {
+        (region.location_id, variable): metrics[region][variable]
+        for region in metrics.keys()
+        for variable in metrics[region].keys()
+    }
+
+    # Make sure there is only one len among all of loc_var_seq.values(). Make a DatetimeIndex
+    # with that many dates.
+    sequence_lengths = more_itertools.one(set(len(seq) for seq in loc_var_seq.values()))
+    dates = pd.date_range(start_date, periods=sequence_lengths, freq="D", name=CommonFields.DATE)
+
+    index = pd.MultiIndex.from_tuples(
+        loc_var_seq.keys(), names=[CommonFields.LOCATION_ID, PdFields.VARIABLE],
+    )
+
+    df = pd.DataFrame(list(loc_var_seq.values()), index=index, columns=dates)
+    df = df.fillna(np.nan).apply(pd.to_numeric)
+
+    dataset = timeseries.MultiRegionDataset.from_timeseries_wide_dates_df(df, dropna=dropna)
+
+    loc_var_provenance = {
+        key: ts_lit.provenance
+        for key, ts_lit in loc_var_seq.items()
+        if isinstance(ts_lit, TimeseriesLiteral)
+    }
+    if loc_var_provenance:
+        provenance_index = pd.MultiIndex.from_tuples(
+            loc_var_provenance.keys(), names=[CommonFields.LOCATION_ID, PdFields.VARIABLE]
+        )
+        provenance_series = pd.Series(
+            list(loc_var_provenance.values()),
+            dtype="str",
+            index=provenance_index,
+            name=PdFields.PROVENANCE,
+        )
+        dataset = dataset.add_provenance_series(provenance_series)
+
+    return dataset
+
+
+def build_default_region_dataset(
```

Comment:
how would you feel about: 
```
def build_single_region_dataset(metrics: Mapping..., *, region=DEFAULT_REGION, dropna=True)
```

lets you customize the region if you want

Reply:
I'm down for that YAGNI.